bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/util/ThingHelper.java

```diff
@@ -90,10 +95,28 @@ private static String toString(List<Channel> channels) {
     }
 
     public static void addChannelsToThing(Thing thing, Collection<Channel> channels) {
+        List<Channel> copy = new LinkedList<>(thing.getChannels());
+        copy.addAll(channels);
+        ensureUnique(copy);
         ((ThingImpl) thing).getChannelsMutable().addAll(channels);
     }
 
     /**
+     * Ensures that there are no duplicate channels in the list (i.e. not using the same ChannelUID)
+     *
+     * @param channels the {@link List} of channels to check
+     * @throws IllegalArgumentException in case there are duplicate channels found
+     */
+    public static void ensureUnique(List<Channel> channels) {
```

Comment:
You have to distinguish: In ThingBuilder this was done before, to ensure the Thing gets a copy so it cannot be modified. I conveniently use it and only update the Thing with it if it's correct.
In the `ThingHelper. addChannelsToThing()` above it's true, I created that temporary copy. I did that in order to do the check _before_ the actual collection gets modified. There is no value in throwing an IllegalArgumentException if we accept the new Channel anyway...